src/PlaywrightSharp/Worker.cs

```diff
@@ -53,48 +52,35 @@ public Worker(IChannelOwner parent, string guid, WorkerInitializer initializer)
 
         internal BrowserContext BrowserContext { get; set; }
 
-        /// <inheritdoc />
-        public async Task<IJSHandle> EvaluateHandleAsync(string expression)
-            => (await _channel.EvaluateExpressionHandleAsync(
+        public async Task<T> EvaluateAsync<T>(string expression, object arg)
+            => ScriptsHelper.ParseEvaluateResult<T>(await _channel.EvaluateExpressionAsync(
                 script: expression,
                 isFunction: expression.IsJavascriptFunction(),
-                arg: EvaluateArgument.Undefined).ConfigureAwait(false))?.Object;
+                arg: TranslateObject(arg)).ConfigureAwait(false));
 
-        /// <inheritdoc />
         public async Task<IJSHandle> EvaluateHandleAsync(string expression, object arg)
             => (await _channel.EvaluateExpressionHandleAsync(
                 script: expression,
                 isFunction: expression.IsJavascriptFunction(),
-                arg: ScriptsHelper.SerializedArgument(arg)).ConfigureAwait(false))?.Object;
-
-        /// <inheritdoc/>
-        public async Task<T> EvaluateAsync<T>(string expression)
-            => ScriptsHelper.ParseEvaluateResult<T>(await _channel.EvaluateExpressionAsync(
-                script: expression,
-                isFunction: expression.IsJavascriptFunction(),
-                arg: EvaluateArgument.Undefined).ConfigureAwait(false));
+                arg: TranslateObject(arg))
+            .ConfigureAwait(false))?.Object;
 
-        /// <inheritdoc/>
-        public async Task<JsonElement?> EvaluateAsync(string expression)
-            => ScriptsHelper.ParseEvaluateResult<JsonElement?>(await _channel.EvaluateExpressionAsync(
-                script: expression,
-                isFunction: expression.IsJavascriptFunction(),
-                arg: EvaluateArgument.Undefined).ConfigureAwait(false));
+        public async Task<IWorker> WaitForCloseAsync(float? timeout)
+        {
+            using var waiter = new Waiter();
+            var waiterResult = waiter.WaitForEvent(this, nameof(Close));
+            await waiterResult.Task.WithTimeout(Convert.ToInt32(timeout ?? 0)).ConfigureAwait(false);
+            return this;
+        }
 
-        /// <inheritdoc/>
-        public async Task<JsonElement?> EvaluateAsync(string expression, object arg)
-            => ScriptsHelper.ParseEvaluateResult<JsonElement?>(await _channel.EvaluateExpressionAsync(
-                script: expression,
-                isFunction: expression.IsJavascriptFunction(),
-                arg: arg,
-                serializeArgument: true).ConfigureAwait(false));
+        private object TranslateObject(object obj)
```

Comment:
Why do we need this? Can't we use `EvaluateArgument.Undefined` and that's it?

Reply:
We _could_, but it would mean making a whole bunch of the other stuff public, that's currently `internal` and I would much much rather have a separate object. It also doesn't mess with the logic of where it's used (i.e. it's not used for transport, but it gets translated) and that will bring it inline to where we want to be, mid-term future. Hence my decision to go with a separate one.